base_vat_optional_vies/models/res_partner.py

```diff
@@ -0,0 +1,94 @@
+        "for example with `easy_install vatnumber` or "
+        "`pip install vatnumber`.")
+    vatnumber = None
+
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+    vies_passed = fields.Boolean(
+        string="VIES validation passed", readonly=True)
+
+    def _split_vat(self, vat, country=False):
+        """
+        @summary: Split Partner vat into country_code and number
+        @result: (vat_country, vat_number)
+        """
+        vat_country = 'XX'
+        vat_number = vat
+        if vat and re.match(r'[A-Za-z]{2}', vat):
+            vat_country = vat[:2].lower()
+            vat_number = vat[2:].replace(' ', '')
+        elif country:
+            vat_country = country
+        return vat_country, vat_number
+
+    @api.model
+    def simple_vat_check(self, country_code, vat_number):
+        res = super(ResPartner, self).simple_vat_check(
+            country_code, vat_number)
+        if res and self.vies_passed and not self.company_id.vat_check_vies:
+            # Can not be sure that this VAT is signed up in VIES
+            self.vies_passed = False
+        return res
+
+    @api.model
+    def vies_vat_check(self, country_code, vat_number):
+        data = {}
+        res = False
+        test_vies = self.env.context.get('test_vies')
+        try:
+            # Validate against VAT Information Exchange System (VIES)
+            # see also http://ec.europa.eu/taxation_customs/vies/
+            vat = country_code + vat_number
+            if not test_vies and not config['test_enable']:
+                res = vatnumber.check_vies(vat)
+            else:
+                res = (True if test_vies == 'y' else False)
+            if res and not self.vies_passed:
+                data['vies_passed'] = True
+        except Exception:
+            # See:
+            #   http://ec.europa.eu/taxation_customs/vies/checkVatService.wsdl
+            # Fault code may contain INVALID_INPUT, SERVICE_UNAVAILABLE,
+            # MS_UNAVAILABLE, TIMEOUT or SERVER_BUSY. There is no way we can
+            # validate the input with VIES if any of these arise, including
+            # the first one (it means invalid country code or empty
+            # VAT number), so we fall back to the simple check.
+            pass
+        if not res:
+            res = self.simple_vat_check(country_code, vat_number)
+            if self.vies_passed:
+                data['vies_passed'] = False
+        if res:
+            vat = country_code.upper() + vat_number
+            if self.vat != vat:
+                data['vat'] = vat
+        if data:
+            self.with_context(avoid_check_vat=True).write(data)
+        return res
+
+    @api.constrains('vat')
+    def check_vat(self):
+        for partner in self:
+            if (not self.env.context.get('avoid_check_vat') and
+                    not partner.parent_id):
+                super(ResPartner, partner).check_vat()
```

Comment:
I don't get why do you need to still overwrite this if you have already touched the other methods.